reflex-dom-core/src/Reflex/Dom/Builder/Immediate.hs

```diff
@@ -624,13 +639,30 @@ instance SupportsImmediateDomBuilder t m => MountableDomBuilder t (ImmediateDomB
       liftIO $ writeIORef (_immediateDomFragment_state childFragment) $ FragmentState_Mounted (before, after)
     liftIO $ writeIORef (_immediateDomFragment_state fragment) $ FragmentState_Mounted (before, after)
 
-instance (Reflex t, MonadAdjust t m, MonadJSM m, MonadHold t m, MonadFix m, PrimMonad m) => MonadAdjust t (ImmediateDomBuilderT t m) where
+-}
+
+instance Monad m => MonadMountStatus t (ImmediateDomBuilderT t m) where
+  getMountStatus = ImmediateDomBuilderT $ asks _immediateDomBuilderEnv_mountState
+
+-- |Structure which keeps track of which generation of a 'runWithReplace' action is currently installed in the DOM.
+data Cohort = Cohort
+  { _cohort_generation :: !Int
+  -- ^Sequentially assigned generation number for this cohort
+  , _cohort_triggerUpdateMountState :: !(MountState -> IO ())
+  -- ^IO action which updates the '_immediateDomBuilderEnv_mountState' for this generation, so that a previous generation can be informed that it's
+  -- unmounted.
+  }
+
+instance (Reflex t, MonadAdjust t m, MonadJSM m, MonadHold t m, MonadFix m, MonadRef m, Ref m ~ IORef, MonadReflexCreateTrigger t m, PrimMonad m) => MonadAdjust t (ImmediateDomBuilderT t m) where
   runWithReplace a0 a' = do
     initialEnv <- ImmediateDomBuilderT ask
     before <- textNodeInternal ("" :: Text)
     let parentUnreadyChildren = _immediateDomBuilderEnv_unreadyChildren initialEnv
+        parentMountState      = _immediateDomBuilderEnv_mountState      initialEnv
```

Comment:
I generally prefer to not do the 2D alignment of `=` and such, since it forces folks to modify more lines in future diffs, and can be tough for people with less advanced editor setups to work on efficiently.